elements/lisk-p2p/src/utils/validate.ts

```diff
@@ -44,40 +44,53 @@ const IPV4_NUMBER = 4;
 const IPV6_NUMBER = 6;
 
 const validateNetworkCompatibility = (
-	peerInfo: P2PDiscoveredPeerInfo,
+	peerInfo: P2PPeerInfo,
 	nodeInfo: P2PNodeInfo,
 ): boolean => {
-	if (!peerInfo.nethash) {
+	if (!peerInfo.sharedState) {
 		return false;
 	}
 
-	return peerInfo.nethash === nodeInfo.nethash;
+	if (!peerInfo.sharedState.nethash) {
```

Comment:
You can do this check and one above in one block with `||`

Reply:
I separated it because typescript was throwing an error, not only we want to check if `peerInfo.sharedState` exist to check `peerInfo.sharedState.nethash` but also have to check if `peerInfo.sharedState` is undefined.